clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1256,7 +1256,8 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
                         log.debug("Discarding stale fetch response for partition {} since the fetched offset {} " +
                                 "does not match the current offset {}", tp, fetchOffset, subscriptions.position(tp));
                     } else if (subscriptions.hasDefaultOffsetResetPolicy()) {
-                        log.info("Fetch offset {} is out of range for partition {}, resetting offset", fetchOffset, tp);
+                        log.info("Fetch offset {} is out of range for partition {}. We only have log segments in " +
+                                "the range from {} to {}. Resetting offset", fetchOffset, tp, partition.logStartOffset, partition.lastStableOffset);
```

Comment:
`logStartOffset` and `lastStableOffset` are not getting filled in when server responds with OffsetOutOfRange error. As @hachikuji pointed out, you need to fill that in on server side here: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ReplicaManager.scala#L1031